lib/index.ts

```diff
@@ -15,6 +15,28 @@ const cancelableClient: CancelableAxiosInstance = Object.assign(client, {
 	CancelToken: Axios.CancelToken,
 	isCancel: Axios.isCancel,
 })
+cancelableClient.interceptors.response.use(r => r, async (error) => {
+	const { config, response, request: { responseURL } } = error
+	const { status, headers } = response
+	const originalRequest = config
+
+	if (status === 503
+		&& headers['x-nextcloud-maintenance-mode'] === '1'
+		&& config.retryIfMaintenanceMode
+		&& (!config._retries || config._retries < 4)) {
```

Comment:
Can we configure `_retries` ?

Reply:
And in that case, maybe use exponential delay?

Reply:
No, this is just an internal state

Reply:
Exponential delay sounds good. Let me give that a try.